### PR TITLE
fix(ui): extend menu highlights to panel edges

### PR DESCRIPTION
**Pull request by rndteams-habbibi**
## Summary
- remove the flat anchored menu's extra vertical content padding
- let the first and last option highlights meet the popup's clipped top and bottom edges
- add a widget regression test for both edge positions

## Test plan
- [x] `flutter test test/components/prego_anchor_menu_test.dart` (module_prego)
- [x] `flutter test test/features/project_list/project_tile_menu_test.dart test/features/session_list/session_tile_menu_test.dart` (app)
- [x] `flutter analyze` (module_prego)

Made with [Cursor](https://cursor.com)

---
## Summary by cubic
Extend flat anchored menu item highlights to the panel’s top and bottom edges. Keep custom/structural entries inset while relying on the panel clip for smooth outer corners.

- **Bug Fixes**
  - Removed outer vertical padding in `PregoAnchorMenu` only when the first/last entries are items; custom or structural edge entries keep a 6px inset.
  - Set menu row `InkWell` borderRadius to zero so highlights meet the panel clip; added tests for edge item bounds and custom-entry insets.

<sup>Written for commit 19303915c187e7a5af8bccf9c288e5f6ebb9113f. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/sesori-ai/sesori_apps_monorepo/pull/452?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by rndteams-habbibi** [comment]
Verified on an iPhone Air simulator running iOS 26.5. This captures the project-row long-press menu while holding the final **Hide Project** action; its pressed highlight now reaches the popup's clipped bottom edge.

Project long-press menu on iOS simulator

File: client/module_prego/lib/components/menus/prego_anchor_menu.dart [hunk 1]
```diff
@@ -311,18 +311,15 @@ class _PregoAnchorMenuState extends State<PregoAnchorMenu> {
...
-          ],
-        ),
+      // AnchoredFlatPanel owns the scroll and clips this column to the panel's
+      // rounded shape. Keeping the column flush with that clip lets the first
+      // and last item highlights reach the menu edges.
+      childBuilder: (context, close) => Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          for (final entry in widget.entriesBuilder()) _flatEntry(context, entry: entry, close: close),
```
[the remark this answers is not in the document]
Comment (rndteams-habbibi) [reply client/module_prego/lib/components/menus/prego_anchor_menu.dart]: Addressed in 19303915. Flat item ink now uses `BorderRadius.zero`, leaving the enclosing panel Material as the single clipping authority for its configured radius. The widget test also asserts the item ink radius and edge geometry.

File: client/module_prego/lib/components/menus/prego_anchor_menu.dart [hunk 2]
```diff
@@ -311,18 +311,15 @@ class _PregoAnchorMenuState extends State<PregoAnchorMenu> {
...
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: [
-            for (final entry in widget.entriesBuilder()) _flatEntry(context, entry: entry, close: close),
-          ],
-        ),
+      // AnchoredFlatPanel owns the scroll and clips this column to the panel's
+      // rounded shape. Keeping the column flush with that clip lets the first
+      // and last item highlights reach the menu edges.
+      childBuilder: (context, close) => Column(
```
[the remark this answers is not in the document]
Comment (rndteams-habbibi) [reply client/module_prego/lib/components/menus/prego_anchor_menu.dart]: Addressed in 19303915. Edge padding is now conditional: custom, label, and divider entries retain the original 6 px inset when they bookend the menu, while a first or last `PregoMenuItem` stays flush. Added a regression test for custom entries at both panel edges.

[the remark this answers is not in the document]
Comment (rndteams-habbibi) [reply client/module_prego/lib/components/menus/prego_anchor_menu.dart]: Addressed in 19303915. Non-item entries now keep the original 6 px edge inset, while only item rows meet the panel edge. The new widget test covers custom entries at both the top and bottom boundaries.
